### DISCUSSION TITLE
Folders or grouping pages in GitHub Wikis

### DISCUSSION THREAD

**Discussion by ghost** [community/community#23914] [New to GitHub]
<p>Hi!</p>
<p>I was working on translating the documentation of a project and I stumble in a problem. How to organize documentation of diferent languages in a wiki? Is there a way to group pages in folders or in collapsable groups?</p>
<p>As far as I see, all the pages get placed in the same “level”.</p>
<p>Any ideas? thanks in avance.</p>
<p>Dknacht</p>

**Comment by ghost** [comment, marked as answer]
<p>Hi <a class="mention" href="/dknacht">@dknacht</a>,</p>
<p>While there currently isn’t a way to group pages by folder or in collapsible groups, you could <a href="https://help.github.com/articles/creating-a-sidebar/" rel="nofollow noopener">set up a custom sidebar</a> to achieve a similar effect which would allow you to link to each Wiki page in categorised lists. It might also be worth <a href="https://github.com/contact" rel="nofollow noopener">contacting GitHub Support</a> to note this as a feature request.</p>
<p>Would something like that work for you? ​If there’s anything else I can help you with, let me know; so happy to help further!</p>
<p>Kindest regards,<br>
Matthew</p>

**Reply by tormung83** [reply]
The answer is simple: GitHub does not support this feature and it won't support it anytime soon. I got a response from Support stating this is not in their roadmap. 
I will cancel my Pro subscription for now but continue using the free version and move my project to [GitBook](https://www.gitbook.com/) as it offers what GitHub can't right now. 
Then I will just integrate it with GitHub. Here are some tutorials that convinced me to move to GitBook.
[(18) GitBook - Getting started - YouTube](https://www.youtube.com/watch?v=-MdyiRH6OAc)
[(18) GitBook - GitHub integration - YouTube](https://www.youtube.com/watch?v=d4s0Ks0e-tA)

**Reply by pythoninthegrass** [reply]
Good find @tormung83. Seems like such a basic feature and disappointing that we have to resort to third-party sites to implement it.

**Comment by ghost** [comment]
<p>Hi! <a class="mention" href="/matthewheath">@matthewheath</a><br>
Thanks so much for your answer.<br>
I will check the custom sidebars, it looks like they are going to do the job.<br>
In <a>this page</a> the image shows pages listed in embedded lists. Do you know how to make this?</p>
<p>Thanks so much again and best regards.<br>
Dknacht.</p>

**Comment by ghost** [comment]
<p>Hello <a class="mention" href="/dknacht">@dknacht</a>,</p>
<p>Sure — they’re just HTML code using list elements.</p>
<p>The actual code would look something like</p>
<pre><code>&lt;ul&gt;
  &lt;li&gt;Top level page
    &lt;ul&gt;
      &lt;li&gt;Sub-page&lt;/li&gt;
      &lt;li&gt;Sub-page two&lt;/li&gt;
    &lt;/ul&gt;
&lt;/li&gt;
&lt;/ul&gt;
</code></pre>
<p>That would then give you a nested list like the image shows.</p>

**Comment by ghost** [comment]
<p>Great!,</p>
<p>That’s what I needed.</p>
<p>I am going to conact github support tho so see if they can implement something.</p>
<p>Thanks man!</p>
<p>Have a good day!.</p>
<p>Dknacht</p>

**Comment by berto** [comment]
<p>Please implement this! </p>
<p>I’m sendin a message to support as well.</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/dknacht">@dknacht</a> and <a class="mention" href="/tallberto">@tallberto</a>,</p>
<p>Thanks for this feedback! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. I’ve logged your feature request in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>
<p>Please let me know if you have any other questions.</p>
<p>Thanks!</p>

**Reply by tormung83** [reply]
Hello, @ThirdEyeRose did this feature ever made it to the discussion table? I personally need to work on a Wiki project and a Side bar is not really what I'm really looking for. I need to place notes (pages) inside folders or collapsable pages.

**Reply by ThirdEyeRose** [reply]
Unfortunately, I do not know. I left GitHub a few years ago. During that time, I had very little insight into the product roadmap as a support person.

**Comment by rusioana** [comment]
<p>Hi, where exactly do you add the HTML formating for the list? In the first page of the section which you want to create?</p>

**Comment by ghost** [comment]
<p>I’m trying to do this too and I cannot get it to work, I’ve googled it extensively and found a couple of links, specifically <a href="https://github.com/tajmone/github-tests/wiki/subpage,%C2%A0https://stackoverflow.com/questions/11088285/github-wiki-directories" rel="nofollow noopener">https://github.com/tajmone/github-tests/wiki/subpage, https://stackoverflow.com/questions/11088285/github-wiki-directories</a> and <a href="https://github.com/KirstieJane/STEMMRoleModels/wiki/Creating-new-folders-in-GitHub-repository-via-the-browser" rel="nofollow noopener">https://github.com/KirstieJane/STEMMRoleModels/wiki/Creating-new-folders-in-GitHub-repository-via-the-browser</a></p>
<p>When I try it, I don’t get anything like a subfolder, just a new file with the path as it’s name and the forward slashes replaced with space, like this</p>
<p>subfolder subpage.md</p>
<p>When I try creating the sub sidebar it just overwrites the original sidebar and the Create Custom Sidebar dissapears, which is not what I want.</p>
<p>I know this can be done because I’ve seen several github wikis which do this. But I don’t know exactly how. Please help, going slightly nuts here</p>

**Comment by demisx** [comment]
<p>Please implement this feature. GitLab Wiki does this and it’s really useful</p>

**Comment by bisdevres** [comment]
please implement this.

i also contacted support for this

**Comment by demisx** [comment]
We need this feature desperately. Maintaining Wiki with more than a few pages is very impractical under current functionality. Thank you. 🙏🏻

**Comment by abalter** [comment]
Adding my support.

**Comment by TonyWhiteSMS** [comment]
Having to replace a Confluence site with GitHub wiki as business won't pay for enough licences to make Confluence accessible by those that need it.  Folders and grouping really are needed

**Comment by jakurll** [comment]
PLEASE add this :)

**Comment by knassre-cmu** [comment]
Pretty please add this feature :)

**Comment by giuliano-97** [comment]
Please add this feature :)

**Comment by ArmenHakobyanEpam** [comment]
+1 for adding this feature, need it badly )

**Comment by alimttth** [comment]
<ul>
  <li>Top level page
    <ul>
      <li>Sub-page</li>
      <li>Sub-page two</li>
    </ul>
</li>
</ul>

**Comment by swftvsn** [comment]
Pretty please?

**Reply by swftvsn** [reply]
Asking for the feature to create tree like structure of wiki pages.

**Comment by bisdevres** [comment]
i have the feeling that Github is not developing the wiki functionality.

our GitHub consultant suggested we should use GitHub pages to publish our wiki.
Apparently their own wiki is also published as GitHub pages.

**Comment by tsmcgrath** [comment]
+ 1 for adding a folder structure to Wiki pages.

**Reply by dmccoystephenson** [reply]
This feature would be useful to me.

**Reply by jimrazmus** [reply]
I would appreciate this functionality as well.

**Comment by hoppaman** [comment]
Wikipedia side of github is completely broken. This issue describes feature it really needs. Setting up and managing side bar by hand and linking own pages is no option.

I also note that github provides support to edit the pages with git repository. I tried that and since my pages use lot of characters that are problem for git like ":" etc.. the whole git repository exploded when I checked it out.

Please make this feature useable. I pay for github monthly ( not on this account ) and I expect better service.

**Comment by ptmcg** [comment]
As an alternative, if you embed h2 Markdown headers in your page, they will be displayed as expandable/collapsable list items in the sidebar (instead of creating a hokey HTML nested list of links).

**Comment by RdeWilde** [comment]
Implement it right now or deprecate the whole wiki functionality

**Reply by dmccoystephenson** [reply]
That seems drastic.

**Reply by RdeWilde** [reply]
> That seems drastic.

If five years to come up with a decision is drastic, I like drastic

**Reply by dmccoystephenson** [reply]
fair lol

**Reply by DIYsciBorg** [reply]
Trying to migrate our hackerspace's wiki and coming up against this wall.. its pretty much $hi7 that a wiki can't support paths.. like.. what year is it?  Doing ABSOLUTELY NOTHING to address the issue???

**Reply by dmccoystephenson** [reply]
What's your hackerspace about? @DIYsciBorg

**Reply by DIYsciBorg** [reply]
we are Tokyo HackerSpace  https://github.com/TokyoHackerspace

**Reply by DIYsciBorg** [reply]
Heres the wiki Im working on https://github.com/TokyoHackerspace/Tokyo-HackerSpace-documentation/wiki

**Reply by DIYsciBorg** [reply]
Where the lack of paths is really going to hurt (and make me rethink the whole thing) is the tools and equipment docs.. we have two laser cutters, three 3d printers, a CNC machine, a leather sewing machine, welding equipment... etc..  A custom sidebar will _work_ but not exactly be ideal or easily managed.

**Comment by ptmcg** [comment]
Unsubscribed - there is no new activity on this issue, only more people asking for this feature.

**Comment by dmccoystephenson** [comment]
Remaining subscribed -- I love checking back on this discussion

**Comment by QuintinWillison** [comment]
I just tried creating a GitHub Wiki, then cloned it locally to discover the default branch name is still `master` - that is perhaps the most disappointing thing for me despite all the efforts taken by GitHub to move us all to `main` elsewhere (like, years ago at this point), this Wiki 'product' has clearly been left to rot. And it's switched on for new repositories by default. 😞

**Comment by boilemmashem** [comment]
wiki seems pretty pointless without proper folder structure, would like to use this feature

**Comment by tedwuusa** [comment]
I found the following project that works around this using custom sidebar (although not perfect)

https://github.com/adriantanasa/github-wiki-sidebar

You can see an example here:
https://github.com/angular/angular.js/wiki

**Comment by paulz** [comment]
Wiki has grown on our project, would love to have nested pages to organize.

**Comment by TheDuckCow** [comment]
Also voicing in that I started using Pages in one project, ran into this, and now it feels stunted. I'm actively avoiding using Pages for other projects solely for this reason.

I'm sure your product managers look at the existing adoption of a suite of features (Pages) as a measure to indicate how many users would benefit from a new feature (nesting pages/folders) - but I implore you think of the negative adoption induced by this missing feature, overall adoption of Pages would surely rise!

**Comment by OneEyedSi** [comment]
A suggestion: Allow users to create a folder structure or sub-pages via forward slashes in the page name.

eg page names 
"Reference/Function1"
"Reference/Function2"

would create a folder or parent page named "Reference" with two sub-pages, "Function1" and "Function2".

Azure Blob Storage uses this concept to create storage folders.  I think it's an elegant and simple approach, easy for users to understand.  And no doubt easier to code up that some sort of drag and drop functionality, like Confluence has.

**Reply by uogbuji** [reply]
Yeah, this slash approach is how GitLab implements it, and I was rather stunned to see GitHub doesn't have this feature, and come across this thread.

**Comment by mstf66x** [comment]
We need this feature!

**Comment by ciphrd** [comment]
+1
It's not even a QOL improvement, not having this makes documentation for any project needing 10+ pages unusable

**Reply by tsmcgrath** [reply]
100%. I literally had this decision to make and using the Github Wiki without hierarchy was unusable and we did not use it. I would use it if it were hierarchical.

**Comment by Albert7866** [comment]
In many wiki platforms, you can organize different language versions of documentation using namespaces or categories. Create separate namespaces or categories for each language to group pages accordingly. Some platforms also support collapsible sections or templates for language navigation. Check your [wiki](https://gloryshinestore.com/) platform's documentation for specific features and guidance on organizing multilingual content.

**Reply by lwestfall** [reply]
cool story chatgpt

**Comment by Albert7866** [comment]
Most wikis allow organizing pages using categories or tags. Utilize these features to group pages by language. Alternatively, consider using templates or naming conventions to distinguish between languages within the page [titles](https://gloryshinestore.com/).

**Reply by tsmcgrath** [reply]
This is orthogonal to the hierarchy feature request, though still a great idea.

**Comment by LinjingZhang** [comment]
It would be great to have this feature, otherwise using Wiki to write and manage documentation is a nightmare for everybody...

**Comment by 269652** [comment]
+1 how is this still an issue? ...

**Comment by XavierEymard** [comment]
I dunno if my answer is appropriate but you can numbering your pages with 1, 1.1, 1.2, 1.2.1 - blabla. it will create an unlimited hierarchy. BUT you won't be able to exceed the 0 to 9 main chapters with this numbering plan. Hope it helps.

**Comment by joaoluiz-tw-em** [comment]
And here we are again. ⏳

**Comment by binkley** [comment]
I tried an experiment of:
```shell
cd $MY_WIKI_REPO
mkdir test
echo '# Test wiki organizing' >test/Test.md
git add .
git commit -m 'Testing wiki organizing'
git push
```
I can find my "Test" page in the "Pages" droplist in the navbar, however there is no directory structure preserved, no mention of the "test" directory. This one simple thing would immediately open to me ways of structuring the large GitHub wiki I work on. Presently we fallback on edits to `_Sidebar.md` and a `Home.md` page with a handmade table of contents.

**Comment by davidvc** [comment]
![image](https://github.com/community/community/assets/71551/d4b0baee-7f52-4a99-99f0-bfb0ff1a92b4)

**Comment by Arcanorum** [comment]
Looks like all the people who worked on this at GH were hit by the layoffs 😐

**Comment by HotNoob** [comment]
y no github implement a needed and low maintence feature that can be added in 5 minutes?

**Comment by Waterstraal** [comment]
+1 

Is this on a roadmap item I can track? Or is this feature not planned at all? I would be interested to know before I migrate my wiki over. 

Thanks

**Comment by 1543siddhant** [comment]
Hi Dknacht,

Organizing documentation for multiple languages in a wiki can be effectively managed through structured organization and clear navigation. While many wikis may initially place all pages on the same level, there are several strategies you can employ to improve organization and make navigation more intuitive:

Use Categories or Tags: Most wikis support categorization or tagging of pages. Create categories or tags for each language (e.g., English, Spanish, French) and assign relevant tags to each page. Users can then filter or search by language.

Namespace or Prefix: Some wikis allow you to create namespaces or use prefixes in page titles. For instance, you might prepend each page title with the language code (e.g., en:Page_Title for English, es:Page_Title for Spanish). This keeps pages organized by language and facilitates quick identification.

Table of Contents: Create a master table of contents page where you list all documentation sections for each language. This acts as a central hub linking to different language-specific sections or pages.

Page Structure: Within each language section, maintain a consistent structure for pages. For example, use the same headings and subheadings to cover similar topics across languages. This makes it easier for users to navigate and find information.

**Reply by Arcanorum** [reply]
"As an AI language model..."

**Comment by LiteBrite82** [comment]
Hi @1543siddhant 

We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.

**Comment by tsmcgrath** [comment]
Categories, Tags, Namespaces, used to replace hierarchies all are hacks that do not replace the core functionality of a folder hierarchy that 1) organizes and 2) hides non-relevant pages. Table of Contents should be automatically generated from folders (like a chapter) and heading level. This is a core feature of other Wikis and makes them functional for any organization with more than one individual. Without it, this is a drastically neutered feature. I live with the neutered feature set of Github Projects but don't like it. I cannot use the neutered feature set of Github Wikis.

**Comment by carlitoplatanito** [comment]
Just curious, if you clone the wiki and edit it that way - even if not displayed on GitHub correctly, the folders are tech still in the repo when/if others clone it as well right?

**Comment by dxm1244** [comment]
It's an embarrassment that GitHub didn't have this feature SIX years ago when this issue was created. Even more now.

**Comment by JonathanDHolden** [comment]
I've just paid for GitHub wiki, only to discover that it's unusable as there are no folders. 

Looks like it's time for a chargeback.

**Comment by gabezurita** [comment]
The need for Github WIki folders has been a request for at least 12 years:
https://stackoverflow.com/questions/11088285/github-wiki-directories

Here is some additional history:
https://github.com/orgs/community/discussions/14020

**Comment by HabibaGadalla** [comment]
still no proper wikis structure ):

**Comment by mdtowfik222** [comment]
Hi

**Comment by mdtowfik222** [comment]
I am YouTube content creator, My GitHub Account suspended! but Why?

**Comment by clombardi** [comment]
I'm so surprised that GitHub Wiki lacks some basic features so it becomes impossible to properly organize the contents. Currently I'm locked on for corporative reasons, but the outcome I can produce is quite sad to see.

**Comment by singhsoldier13** [comment]
+1

**Comment by MackHalliday** [comment]
why isn't this supported? Such a simple feature.

**Reply by ccgargantua** [reply]
because microsoft hates their users and only delivers to shareholders

**Comment by nurbek-mirai** [comment]
Bump.

**Comment by edgar-kautzner** [comment]
Please develop this feature - this is really needed!!

**Comment by pmatulis** [comment]
I am veritably gobsmacked that you cannot use the Wiki feature as... a wiki.

**Comment by dazzknowles** [comment]
+1 Here too, this is basic functionality that is sorely lacking.

**Comment by smith-sj** [comment]
I just started using GitHub's Wiki feature and am disappointed such a basic functionality is missing. Adding this to the list of things Azure DevOps (surprisingly) does much better.

**Comment by thomaswitt** [comment]
+1 - wikis with subdirectories would be a great feature. This way one could integrate docs like PRDs for a project as git submodule and have a comfortable way to view and edit them.
